### DISCUSSION TITLE
buffer problem

### DISCUSSION THREAD

**Discussion by mgardenal** [arkypita/LaserGRBL#1830] [Q&A]
Hi friends, I'm new here..
I'm trying to convert a chinese wainlux jl1 machine because it burned the board
It's already working with GRBL, but it has a lot of hard limit problems and unexpected stops during recording because of buffering
LaserGrbl always starts blocked with ALARM warning, is this normal?
I've already adjusted her limit to the size of the table, I've already made some recommended settings in other forum topics, but nothing solves.

<HTML>
<BODY>
<TABLE><TR><TD ALIGN="center">&nbsp;</TD><TD>$0</TD><TD>Step pulse time</TD><TD>10</TD><TD>microseconds</TD><TD>Sets time length per step. Minimum 3usec.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$1</TD><TD>Step idle delay</TD><TD>25</TD><TD>milliseconds</TD><TD>Sets a short hold delay when stopping to let dynamics settle before disabling steppers. Value 255 keeps motors enabled with no delay.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$2</TD><TD>Step pulse invert</TD><TD>0</TD><TD>mask</TD><TD>Inverts the step signal. Set axis bit to invert (00000ZYX).</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$3</TD><TD>Step direction invert</TD><TD>0</TD><TD>mask</TD><TD>Inverts the direction signal. Set axis bit to invert (00000ZYX).</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$4</TD><TD>Invert step enable pin</TD><TD>0</TD><TD>boolean</TD><TD>Inverts the stepper driver enable pin signal.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$5</TD><TD>Invert limit pins</TD><TD>0</TD><TD>boolean</TD><TD>Inverts the all of the limit input pins.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$6</TD><TD>Invert probe pin</TD><TD>0</TD><TD>boolean</TD><TD>Inverts the probe input pin signal.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$10</TD><TD>Status report options</TD><TD>1</TD><TD>mask</TD><TD>Alters data included in status reports.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$11</TD><TD>Junction deviation</TD><TD>0,010</TD><TD>millimeters</TD><TD>Sets how fast Grbl travels through consecutive motions. Lower value slows it down.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$12</TD><TD>Arc tolerance</TD><TD>0,002</TD><TD>millimeters</TD><TD>Sets the G2 and G3 arc tracing accuracy based on radial error. Beware: A very small value may effect performance.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$13</TD><TD>Report in inches</TD><TD>0</TD><TD>boolean</TD><TD>Enables inch units when returning any position and rate value that is not a settings value.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$20</TD><TD>Soft limits enable</TD><TD>0</TD><TD>boolean</TD><TD>Enables soft limits checks within machine travel and sets alarm when exceeded. Requires homing.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$21</TD><TD>Hard limits enable</TD><TD>1</TD><TD>boolean</TD><TD>Enables hard limits. Immediately halts motion and throws an alarm when switch is triggered.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$22</TD><TD>Homing cycle enable</TD><TD>1</TD><TD>boolean</TD><TD>Enables homing cycle. Requires limit switches on all axes.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$23</TD><TD>Homing direction invert</TD><TD>2</TD><TD>mask</TD><TD>Homing searches for a switch in the positive direction. Set axis bit (00000ZYX) to search in negative direction.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$24</TD><TD>Homing locate feed rate</TD><TD>25,000</TD><TD>mm/min</TD><TD>Feed rate to slowly engage limit switch to determine its location accurately.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$25</TD><TD>Homing search seek rate</TD><TD>500,000</TD><TD>mm/min</TD><TD>Seek rate to quickly find the limit switch before the slower locating phase.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$26</TD><TD>Homing switch debounce delay</TD><TD>250</TD><TD>milliseconds</TD><TD>Sets a short delay between phases of homing cycle to let a switch debounce.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$27</TD><TD>Homing switch pull-off distance</TD><TD>1,000</TD><TD>millimeters</TD><TD>Retract distance after triggering switch to disengage it. Homing will fail if switch isn't cleared.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$30</TD><TD>Maximum spindle speed</TD><TD>255</TD><TD>RPM</TD><TD>Maximum spindle speed. Sets PWM to 100% duty cycle.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$31</TD><TD>Minimum spindle speed</TD><TD>0</TD><TD>RPM</TD><TD>Minimum spindle speed. Sets PWM to 0.4% or lowest duty cycle.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$32</TD><TD>Laser-mode enable</TD><TD>1</TD><TD>boolean</TD><TD>Enables laser mode. Consecutive G1/2/3 commands will not halt when spindle speed is changed.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$100</TD><TD>X-axis travel resolution</TD><TD>20,000</TD><TD>step/mm</TD><TD>X-axis travel resolution in steps per millimeter.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$101</TD><TD>Y-axis travel resolution</TD><TD>20,000</TD><TD>step/mm</TD><TD>Y-axis travel resolution in steps per millimeter.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$102</TD><TD>Z-axis travel resolution</TD><TD>250,000</TD><TD>step/mm</TD><TD>Z-axis travel resolution in steps per millimeter.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$110</TD><TD>X-axis maximum rate</TD><TD>3000,000</TD><TD>mm/min</TD><TD>X-axis maximum rate. Used as G0 rapid rate.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$111</TD><TD>Y-axis maximum rate</TD><TD>3000,000</TD><TD>mm/min</TD><TD>Y-axis maximum rate. Used as G0 rapid rate.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$112</TD><TD>Z-axis maximum rate</TD><TD>500,000</TD><TD>mm/min</TD><TD>Z-axis maximum rate. Used as G0 rapid rate.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$120</TD><TD>X-axis acceleration</TD><TD>50,000</TD><TD>mm/sec^2</TD><TD>X-axis acceleration. Used for motion planning to not exceed motor torque and lose steps.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$121</TD><TD>Y-axis acceleration</TD><TD>50,000</TD><TD>mm/sec^2</TD><TD>Y-axis acceleration. Used for motion planning to not exceed motor torque and lose steps.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$122</TD><TD>Z-axis acceleration</TD><TD>10000,000</TD><TD>mm/sec^2</TD><TD>Z-axis acceleration. Used for motion planning to not exceed motor torque and lose steps.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$130</TD><TD>X-axis maximum travel</TD><TD>380,000</TD><TD>millimeters</TD><TD>Maximum X-axis travel distance from homing switch. Determines valid machine space for soft-limits and homing search distances.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$131</TD><TD>Y-axis maximum travel</TD><TD>370,000</TD><TD>millimeters</TD><TD>Maximum Y-axis travel distance from homing switch. Determines valid machine space for soft-limits and homing search distances.</TD></TR><TR><TD ALIGN="center">&nbsp;</TD><TD>$132</TD><TD>Z-axis maximum travel</TD><TD>1000,000</TD><TD>millimeters</TD><TD>Maximum Z-axis travel distance from homing switch. Determines valid machine space for soft-limits and homing search distances.</TD></TR></TABLE>

</BODY>
</HTML>

<a href="https://freeimage.host/i/XUS4vS"><img src="https://iili.io/XUS4vS.th.jpg" alt="XUS4vS.th.jpg" border="0"></a>

**Comment by StuartB4** [comment]
Hard limits were really intended for expensive CNC machines to stop the tool smashing in to the machine and causing ££££/$$$$ worth of damage. 
They are not really needed on a laser engraver. I would disable $21 and enable $20.
Soft limits are ok on their own, just make sure you set the size of the engraving area in grbl configuration correctly. 
$130 and $131. 

I have mine set for a few mm's less. My X engraving size is 390 but I set the $130 to 385, Y is 290 but $131 is set to 285.

**Reply by mgardenal** [reply]
Hello, 
the value of $20 and 21 are like this because it was a recommendation from another topic, I used it the other way around, and I use end-of-course buttons, but if they are activated, they are also buffered
the table values are set correctly, this work wue I try to do this in the center of the table and it is 74mm in size, and even so it shows hard limit and buffer with 1 minute of work

**Comment by StuartB4** [comment]
Sometimes interference in the leads, especially the stepper motor leads can cause false triggering of the limit switches, usually when hard limits is enabled. I used to have the same problem but have a main control board with resistor and capacitor filtering, that stops it.

**Reply by mgardenal** [reply]
does this happen even if the motor wires are insulated from each other?  
what about the buffer turning red?

**Comment by StuartB4** [comment]
You could try grounding the metal parts of the frame and the metal parts of the motors.
Ortur lasers have similar problems and now make a grounding kit for them. 

Also try changing from Buffered to Synchronous in the settings on the Grbl menu. 

![image](https://user-images.githubusercontent.com/16980026/171311524-11827de9-f708-4390-acfe-a9281de62b18.png)

**Reply by mgardenal** [reply]
I did what you recommended, I managed to do some tests, it's working but sometimes the machine stops in the middle of the process and doesn't continue at all, you need to close the program and start again.

**Reply by ettore-bernardi** [reply]
You saved me, thanks
